### DISCUSSION TITLE
Wellness reminders

### DISCUSSION THREAD

**Discussion by poperigby** [InfiniTimeOrg/InfiniTime#1598] [Ideas]
### 

- [X] I searched for similar feature request and found none was relevant.

### Pitch us your idea!

I think InfiniTime should be able to periodically send reminders to aid you in your overall wellness, some of them depending on your activity

### Description

Some examples:
- If it detects that you have been sitting down for a while, it should remind you to stand up
- It should remind you at regular intervals to take a walk, so you can attempt to meet your step goal
- Remind you to drink so water
- Remind you to take some deep breaths

The Apple Watch does things like this, so it could be used for reference.

They should probably be toggle-able in a 'Wellness Reminders' menu in the settings.

**Comment by zoltan-kiss-hungary** [comment]
I would like to upvote this, especially the first bulletpoint. Like nearly every office worker, I'm spending too much time sitting, and I often notice that only when it's quite late, and my legs are cramping. I bought a Fitbit, thinking that idle time tracking and notification is a basic use case for fitness trackers,and hence it's richly configurable, but it seems it's not. And neither for other trackers I've checked so far, in fact. The Fitbit accumulate data in every hour between :00 and :50, and if there was not enough movement, there is a gentle vibrating reminder. But if you miss that, there is no new reminder for 60 minutes. Plus if you do your 250 steps e.g. at 8:01am, and then you sit down,there is no reminder until 9:50. It's easy to see this is quite flawed.
I'm willing to work on this, but before I spend a lot of time studying the code, I would like to ask people who are more familiar with it: is the following is possible with PineTime? :
- wake up every 3 minutes, and check how much movement were made in the past 30 minutes (the numbers here should be configurable)
- if not enough (again, number of steps is configurable), vibrate
- do this only in the configurable time window, your work time
Could someone with a good overview give me a first hunch estimation whether this is feasible or not?

**Comment by ghost** [comment]
I don't have quite the insight yet, but as far as I know writing data to external flash and saving steps/HR periodically with a timestamp is currently being worked at.

Have a look at issue/discussion #749 for example
